code/mob/living.dm

```diff
@@ -1502,10 +1504,33 @@ var/global/icon/human_static_base_idiocy_bullshit_crap = icon('icons/mob/human.d
 
 	if (pushpull_multiplier != 0) // if we're not completely ignoring pushing/pulling
 		if (src.pulling)
-			. *= (pull_speed_modifier(move_target) * pushpull_multiplier)
+			if (istype(src.pulling, /atom/movable) && !(src.is_hulk() || (src.bioHolder && src.bioHolder.HasEffect("strong"))))
+				var/atom/movable/A = src.pulling
+				// hi grayshift sorry grayshift
+				if (get_dist(src,A) > 0 && get_dist(move_target,A) > 0) //i think this is mbc dist stuff for if we're actually stepping away and pulling the thing or not?
+					if(pull_slowing)
+						. *= max(A.p_class, 1)
+					else
+						if(istype(A,/obj/machinery/nuclearbomb)) //can't speed off super fast with the nuke, it's heavy
+							. *= max(A.p_class, 1)
+						// else, ignore p_class*/
+						else if(ismob(A))
+							var/mob/M = A
+							//if they're lying, pull em slower, unless you have anext_move gang and they are in your gang.
+							if(M.lying)
+								if (src.mind?.gang && (src.mind.gang == M.mind?.gang))
+									. *= 1		//do nothing
+								else
+									. *= lerp(1, max(A.p_class, 1), mob_pull_multiplier) 
+						else if(istype(A, /obj/storage))
+							// if the storage object contains mobs, use its p_class (updated within storage to reflect containing mobs or not)
+							if (locate(/mob) in A.contents)
+								. *= lerp(1, max(A.p_class, 1), mob_pull_multiplier)
+			. = lerp(1, . , pushpull_multiplier)
+
 
 		if (src.pushing && (src.pulling != src.pushing))
-			. *= (max(src.pushing.p_class, 1) * pushpull_multiplier)
+			. *= lerp(1, max(src.pushing.p_class, 1), pushpull_multiplier)
```

Comment:
Didn't recognize this proc, looked into it. This really should be a macro. It's used in so few places I wouldn't even be against you doing in this PR, but if you'd rather not I'll probably create an Issue for it.

Reply:
saner to macroize separately I think